### ISSUE TITLE
Spies can be detected by Hinds even when disguised

### ISSUE THREAD

**Issue by MikeFender** [OpenRA/OpenRA#12409]
Just what the title says. Can't think why this would be.

**Comment by Snurre86** [comment]
Are you talking about a bots hind or a players hind in attack/defend stance?

**Comment by MikeFender** [comment]
Just when I play against the computer. I have disguised spies running
around and then a chopper comes out of their base and guns them down.

**Comment by chrisforbes** [comment]
Yep, AI code doesn't care about the disguise. [That said, an attentive
human player will likely do the same thing, if they have a hind
available...]

**Comment by MikeCaputo** [comment]
I'll second @MikeFender  though - when I play against an AI, my disguised spy will walk right past infantry, pillboxes, tanks, etc. But the Hinds will spot them and rip them to bits. It definitely seems to have different behavior.

**Comment by KOYK** [comment]
i have added a spy unit in my mod in TD, i have the same problem with orcas, they just kill them, disguised or not no matter for the pilots eagle eyes :p

**Comment by Smittytron** [comment]
I'd swap the Balancing tag for an AI one here.

**Comment by pchote** [comment]
This can probably be fixed by adding an extra `unit.EffectiveOwner` check to
https://github.com/OpenRA/OpenRA/blob/7c901c27f10ca594a56abac5fbbd5635fbf53e6e/OpenRA.Mods.Common/AI/HackyAI.cs#L317-L320

**Comment by penev92** [comment]
Like @chrisforbes  said, a human player is likely to do the same if attentive enough, so simply making the AI incapable of dealing with enemy disguised units would be even worse IMO (it will be nice for the players to experiment with a new toy against the AI, but will make the AI even weaker).

If we go with such a check we should at least give it a (random) chance of failing, to better simulate a player not noticing a spy approaching.

**Comment by KOYK** [comment]
The problem is that some times bots(hinds) pick the spy as a target if it is the closest unit it can attack, but a normal player will suspect a spy if it is closer to his base or after he was being infiltrated, and then be on alert, or if the unit uses that player color. 

But lets not forget that they are players that want the AI to be stupid and players that want it to be realistic, so i think if a change like that is to be made it has to be per bot and not to all bots, so the user can pick the gameplay they want.
